### PR TITLE
GH-41015: [JS] [Benchmarking] allow JS benchmarks to run more portably

### LINKED ISSUE #41015: [JS] [Benchmarking] JavaScript benchmarks don't work on Amazon Linux
### Describe the bug, including details regarding any error messages, version, and platform.

Amazon Linux, which is Fedora-based, does not have the `-S` option for the `env` command. So after

- https://github.com/apache/arrow/pull/38500

was merged, we cannot run `yarn perf` on EC2 instances, due to this shebang:

https://github.com/apache/arrow/blob/72d20ad719021c5513620e23a0a65fb724f0e299/js/perf/index.ts#L1

### Component(s)

Benchmarking, JavaScript

### PR DESCRIPTION
**Pull request by austin3dickey**
Fixes #41015. This PR allows users to run `yarn perf` to run the JS benchmarks on more machines; in particular, machines that do not support `env -S` like Amazon Linux.
* GitHub Issue: #41015

### CODE REVIEW

**Comment by austin3dickey** [comment]
@ursabot please benchmark lang=JavaScript

**Review by domoritz — COMMENTED** [review]
Hmm, now we have the same command (`node --no-warnings --loader ts-node/esm/transpile-only`) in two places (the package.json and the ts file). Where else do we call perf from? Could we just change all the calls to run `yarn perf` / `npm run perf`?

**Comment by austin3dickey** [comment]
>Where else do we call perf from? Could we just change all the calls to run yarn perf / npm run perf?

As far as I know, everything that runs benchmarks just uses `yarn perf`.

>Now we have the same command in two places

If this is a problem, perhaps we should remove the shebang then? The trouble with the shebang as written is that using `/usr/bin/env -S` is non-portable. So if we're using `yarn perf` everywhere, it should be fine to remove the shebang.

**Comment by domoritz** [comment]
Sounds good to me. We have a few other places that use the same shebang and I think we should update them as well. 

Rather than yarn, maybe we should use npm run since it's more portable. But that can be a separate change.

**Comment by austin3dickey** [comment]
>We have a few other places that use the same shebang and I think we should update them as well.

Just looked into this briefly and found places in archery that might fail if we remove the shebangs:

https://github.com/apache/arrow/blob/83359d6958273be534f376f40976196c14675c8b/dev/archery/archery/integration/tester_js.py#L27-L30

So we could go through and fix all these, but it might be more trouble than it's worth. Would you mind if I keep this PR scoped to just fixing the benchmarks' portability? 

(For context, I'm trying to move the orchestration of the Arrow merge-commit JS benchmarks off a machine in someone's basement to an EC2 instance. This PR unblocks that work.)

**Comment by domoritz** [comment]
> So we could go through and fix all these, but it might be more trouble than it's worth. Would you mind if I keep this PR scoped to just fixing the benchmarks' portability?

Agreed. So to unblock you, can you run `node --no-warnings --loader ts-node/esm/transpile-only perf/index.ts` from the perf script? Otherwise, sure, let's merge this as is. I approve either way.

**Review by domoritz — APPROVED** [review]

**Comment by austin3dickey** [comment]
I think that might be slightly difficult on the orchestration side. Cool, thanks for the review. 💜 would you mind merging? I don't have write access.

**Pull request merged**
